### PR TITLE
Feature/issue 360 limit ncbi publications

### PR DESCRIPTION
**Pull request by mike-miroliubov**
# Description

## Background

This PR fixed issue #360: for several genes (e.g. BRCA1, KRAS, EGFR) call to endpoint **GET /externaldb/ncbi/gene/{geneId}/get** 
is failing with 500 Internal Server Error. The issue is being reproduced for reference GTF files: Homo_Sapiens.GRCh37.gtf.gz , Homo_Sapiens.GRCh38.gtf.gz.
 The reason is that those genes have huge number of related publications (~5000 for BRCA1) and NCBI API endpoint https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?retmode=json&query_key=1&WebEnv={webEnv} has a limit of 500 UID to return data for.
This PR fixes this by limiting the maximum number of publications fetched for a gene to 500, adding a *retmax=500* parameter to that endpoint call. After all, more than 500 publications is probably too much for a person to consume through Gene Info Pop-up, so it might make sense to limit this amount...

## Changes

PR adds retmax=500 parameter to all calls to `https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?retmode=json&query_key=1&WebEnv={webEnv}` endpoint.

## Acceptance criteria

1. Register h37 reference genome
2. Register Homo_Sapiens.GRCh37.gtf.gz  GTF file.
2. Open NGB browser
2. Go to BRCA1, KRAS, EGFR genes
3. Verify that NCBI info is fetched correctly

# Check list

- [-] Unit tests are provided
- [-] Integration tests are provided (if CLI/API was changed)
- [-] Documentation is updated

### CODE REVIEW

File: server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/ncbi/NCBIDataManager.java [hunk 1]
```diff
@@ -59,8 +61,15 @@
     protected static final String NCBI_SEARCH = "entrez/eutils/esearch.fcgi?";
     protected static final String NCBI_LINK = "entrez/eutils/elink.fcgi?";
     protected static final String RETMODE_PARAM = "retmode";
+    protected static final Integer MAX_RESULTS_PARAM_VALUE = 500;
```
Comment (mzueva) [inline server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/ncbi/NCBIDataManager.java]: I'd suggest to move this param to `catgenome.properties` to allow customisation and set default value to 100.

File: server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/HttpDataManager.java [hunk 2]
```diff
@@ -83,7 +85,7 @@ public String fetchData(String locationStub, ParameterNameValue[] params)
             throws ExternalDbUnavailableException {
 
         final String location = getLocationStub(locationStub, params);
-
+        log.info(location);
```
Comment (mzueva) [inline server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/HttpDataManager.java:88]: Seems to be a debug logging, maybe we don't need it in this PR?

**Review by mzueva — CHANGES_REQUESTED** [review]

File: server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/ncbi/NCBIDataManager.java (hunk 1 above)
Comment (mike-miroliubov) [reply server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/ncbi/NCBIDataManager.java]: Agree, fixed. Now controlled by `externaldb.ncbi.max.results` property, default to 100 and also added startup check to verify it's not more that max allowed 500.

File: server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/HttpDataManager.java (hunk 2 above)
Comment (mike-miroliubov) [reply server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/HttpDataManager.java:88]: Fixed, removed it

**Review by mzueva — APPROVED** [review]

**Pull request merged**